muskathlon/models/survey_user_input.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Quentin Gigon <gigon.quentin@gmail.com>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from odoo import models, api
+
+
+class SurveyUserInput(models.Model):
+    _inherit = 'survey.user_input'
+
+    @api.multi
+    def write(self, vals):
+        """
+        Automatically complete Medical Survey task when user filled it
+        """
+        res = super(SurveyUserInput, self).write(vals)
+        if vals.get('state') == 'done':
+
+            # Search for medical surveys
+            registrations = self.env['event.registration'].search([
+                ('partner_id_id', 'in', self.mapped('partner_id_id')),
```

Comment:
`self.mapped('partner_id_id')` doesn't exit in surveys. It should be `self.mapped('partner_id').ids`

Reply:
Plus you cannot search over related fields I think. So better to put this:
```suggestion
                ('partner_id', 'in', self.mapped('partner_id').ids),
```

Reply:
'partner_id' is not a field of event.registration, so I can't use it in the search, no ?

Reply:
Yes it's a field in event.registration, only it's defined in the core module event of Odoo, not in our custom module. But you can use it, it exists.

Reply:
Ok! thanks